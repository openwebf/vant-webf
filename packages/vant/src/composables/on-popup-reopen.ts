import { inject, InjectionKey, watch } from 'vue';

// eslint-disable-next-line
export const POPUP_TOGGLE_KEY: InjectionKey<() => boolean> = Symbol();

export function onPopupReopen(callback: () => void) {
  const popupToggleStatus = inject(POPUP_TOGGLE_KEY, null);

  if (popupToggleStatus) {
    watch(popupToggleStatus, (show) => {
      if (show) {
        callback();
      }
    });
  }
}
