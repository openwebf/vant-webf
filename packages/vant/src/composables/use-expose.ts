import { getCurrentInstance } from 'vue';
import { extend } from '../utils';

// expose public api
export function useExpose<T = Record<string, any>>(apis: T) {
  const instance = getCurrentInstance();
  if (instance) {
    extend(instance.proxy as object, apis);
  }
}
