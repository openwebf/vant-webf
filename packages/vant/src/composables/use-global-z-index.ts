/**
 * The z-index of Popup components.

 * Will affect this components:
 *   - ActionSheet
 *   - Calendar
 *   - Dialog
 *   - DropdownItem
 *   - ImagePreview
 *   - Notify
 *   - Popup
 *   - Popover
 *   - ShareSheet
 *   - Toast
 */
let globalZIndex = 2000;

/** the global z-index is automatically incremented after reading */
export const useGlobalZIndex = () => ++globalZIndex;

/** reset the global z-index */
export const setGlobalZIndex = (val: number) => {
  globalZIndex = val;
};
