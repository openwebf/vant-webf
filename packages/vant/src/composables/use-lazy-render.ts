import { ref, watch, WatchSource } from 'vue';

export function useLazyRender(show: WatchSource<boolean | undefined>) {
  const inited = ref(false);

  watch(
    show,
    (value) => {
      if (value) {
        inited.value = value;
      }
    },
    { immediate: true }
  );

  return (render: () => JSX.Element) => () => inited.value ? render() : null;
}
