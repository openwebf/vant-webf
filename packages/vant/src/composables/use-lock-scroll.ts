import { Ref, watch, onBeforeUnmount, onDeactivated } from 'vue';
import { getScrollParent, onMountedOrActivated } from '@vant/use';
import { useTouch } from './use-touch';
import { preventDefault } from '../utils';

let totalLockCount = 0;

const BODY_LOCK_CLASS = 'van-overflow-hidden';

export function useLockScroll(
  rootRef: Ref<HTMLElement | undefined>,
  shouldLock: () => boolean
) {
  const touch = useTouch();
  const DIRECTION_UP = '01';
  const DIRECTION_DOWN = '10';

  const onTouchMove = (event: TouchEvent) => {
    touch.move(event);

    const direction = touch.deltaY.value > 0 ? DIRECTION_DOWN : DIRECTION_UP;
    const el = getScrollParent(
      event.target as Element,
      rootRef.value
    ) as HTMLElement;
    const { scrollHeight, offsetHeight, scrollTop } = el;
    let status = '11';

    if (scrollTop === 0) {
      status = offsetHeight >= scrollHeight ? '00' : '01';
    } else if (scrollTop + offsetHeight >= scrollHeight) {
      status = '10';
    }

    if (
      status !== '11' &&
      touch.isVertical() &&
      !(parseInt(status, 2) & parseInt(direction, 2))
    ) {
      preventDefault(event, true);
    }
  };

  const lock = () => {
    document.addEventListener('touchstart', touch.start);
    document.addEventListener('touchmove', onTouchMove, { passive: false });

    if (!totalLockCount) {
      document.body.classList.add(BODY_LOCK_CLASS);
    }

    totalLockCount++;
  };

  const unlock = () => {
    if (totalLockCount) {
      document.removeEventListener('touchstart', touch.start);
      document.removeEventListener('touchmove', onTouchMove);

      totalLockCount--;

      if (!totalLockCount) {
        document.body.classList.remove(BODY_LOCK_CLASS);
      }
    }
  };

  const init = () => shouldLock() && lock();

  const destroy = () => shouldLock() && unlock();

  onMountedOrActivated(init);
  onDeactivated(destroy);
  onBeforeUnmount(destroy);

  watch(shouldLock, (value) => {
    value ? lock() : unlock();
  });
}
