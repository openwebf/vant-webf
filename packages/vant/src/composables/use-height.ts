import { useRect } from '@vant/use';
import { Ref, ref, onMounted, nextTick } from 'vue';
import { onPopupReopen } from './on-popup-reopen';

export const useHeight = (
  element: Element | Ref<Element | undefined>,
  withSafeArea?: boolean
) => {
  const height = ref<number>();

  const setHeight = () => {
    height.value = useRect(element).height;
  };

  onMounted(() => {
    nextTick(setHeight);

    // If the element is using safe area, the system will not return the correct height on page load.
    // So we need to wait for the height to be set.
    // https://github.com/vant-ui/vant/issues/10131
    // https://stackoverflow.com/questions/64891541
    if (withSafeArea) {
      for (let i = 1; i <= 3; i++) {
        setTimeout(setHeight, 100 * i);
      }
    }
  });

  // The result of useHeight might be 0 when the popup is hidden,
  // so we need to reset the height when the popup is reopened.
  // IntersectionObserver is a better solution, but it is not supported by legacy browsers.
  // https://github.com/vant-ui/vant/issues/10628
  onPopupReopen(() => nextTick(setHeight));

  return height;
};
