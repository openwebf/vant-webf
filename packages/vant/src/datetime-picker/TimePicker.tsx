import {
  ref,
  watch,
  computed,
  nextTick,
  onMounted,
  defineComponent,
} from 'vue';

// Utils
import {
  pick,
  clamp,
  extend,
  padZero,
  createNamespace,
  makeNumericProp,
} from '../utils';
import {
  times,
  sharedProps,
  pickerInheritKeys,
  proxyPickerMethods,
} from './utils';

// Composables
import { useExpose } from '../composables/use-expose';

// Components
import { Picker, PickerInstance } from '../picker';

const [name] = createNamespace('time-picker');

export default defineComponent({
  name,

  props: extend({}, sharedProps, {
    minHour: makeNumericProp(0),
    maxHour: makeNumericProp(23),
    minMinute: makeNumericProp(0),
    maxMinute: makeNumericProp(59),
    modelValue: String,
  }),

  emits: ['confirm', 'cancel', 'change', 'update:modelValue'],

  setup(props, { emit, slots }) {
    const formatValue = (value?: string) => {
      const { minHour, maxHour, maxMinute, minMinute } = props;

      if (!value) {
        value = `${padZero(minHour)}:${padZero(minMinute)}`;
      }

      let [hour, minute] = value.split(':');
      hour = padZero(clamp(+hour, +minHour, +maxHour));
      minute = padZero(clamp(+minute, +minMinute, +maxMinute));

      return `${hour}:${minute}`;
    };

    const picker = ref<PickerInstance>();
    const currentDate = ref(formatValue(props.modelValue));

    const ranges = computed(() => [
      {
        type: 'hour',
        range: [+props.minHour, +props.maxHour],
      },
      {
        type: 'minute',
        range: [+props.minMinute, +props.maxMinute],
      },
    ]);

    const originColumns = computed(() =>
      ranges.value.map(({ type, range: rangeArr }) => {
        let values = times(rangeArr[1] - rangeArr[0] + 1, (index) =>
          padZero(rangeArr[0] + index)
        );

        if (props.filter) {
          values = props.filter(type, values);
        }

        return {
          type,
          values,
        };
      })
    );

    const columns = computed(() =>
      originColumns.value.map((column) => ({
        values: column.values.map((value) =>
          props.formatter(column.type, value)
        ),
      }))
    );

    const updateColumnValue = () => {
      const pair = currentDate.value.split(':');
      const values = [
        props.formatter('hour', pair[0]),
        props.formatter('minute', pair[1]),
      ];

      nextTick(() => {
        picker.value?.setValues(values);
      });
    };

    const updateInnerValue = () => {
      const [hourIndex, minuteIndex] = picker.value!.getIndexes();
      const [hourColumn, minuteColumn] = originColumns.value;

      const hour = hourColumn.values[hourIndex] || hourColumn.values[0];
      const minute = minuteColumn.values[minuteIndex] || minuteColumn.values[0];

      currentDate.value = formatValue(`${hour}:${minute}`);
      updateColumnValue();
    };

    const onConfirm = () => emit('confirm', currentDate.value);
    const onCancel = () => emit('cancel');

    const onChange = () => {
      updateInnerValue();
      nextTick(() => {
        nextTick(() => emit('change', currentDate.value));
      });
    };

    onMounted(() => {
      updateColumnValue();
      nextTick(updateInnerValue);
    });

    watch(columns, updateColumnValue);

    watch(
      () => [props.filter, props.maxHour, props.minMinute, props.maxMinute],
      updateInnerValue
    );

    watch(
      () => props.minHour,
      () => {
        nextTick(updateInnerValue);
      }
    );

    watch(currentDate, (value) => emit('update:modelValue', value));

    watch(
      () => props.modelValue,
      (value) => {
        value = formatValue(value);

        if (value !== currentDate.value) {
          currentDate.value = value;
          updateColumnValue();
        }
      }
    );

    useExpose({
      getPicker: () =>
        picker.value && proxyPickerMethods(picker.value, updateInnerValue),
    });

    return () => (
      <Picker
        v-slots={slots}
        ref={picker}
        columns={columns.value}
        onChange={onChange}
        onCancel={onCancel}
        onConfirm={onConfirm}
        {...pick(props, pickerInheritKeys)}
      />
    );
  },
});
