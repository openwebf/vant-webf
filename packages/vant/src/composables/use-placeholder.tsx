import { Ref } from 'vue';
import { useHeight } from './use-height';
import type { BEM } from '../utils/create';

export function usePlaceholder(contentRef: Ref<Element | undefined>, bem: BEM) {
  const height = useHeight(contentRef, true);

  return (renderContent: () => JSX.Element) => (
    <div
      class={bem('placeholder')}
      style={{ height: height.value ? `${height.value}px` : undefined }}
    >
      {renderContent()}
    </div>
  );
}
