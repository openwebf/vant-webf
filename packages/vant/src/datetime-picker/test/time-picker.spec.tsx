import TimePicker from '../TimePicker';
import { mount, later, triggerDrag } from '../../../test';
import { ref, reactive } from 'vue';
import { useExpose } from '../../composables/use-expose';

function filter(type: string, options: string[]): string[] {
  const mod = type === 'minute' ? 10 : 5;
  return options.filter((option) => Number(option) % mod === 0);
}

function formatter(type: string, value: string): string {
  return `${value} ${type}`;
}

test('format initial value', () => {
  const wrapper = mount(TimePicker, {
    props: {
      minHour: 22,
      minMinute: 58,
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
});

test('max-hour & max-minute', () => {
  const wrapper = mount(TimePicker, {
    props: {
      modelValue: '23:59',
      maxHour: 2,
      maxMinute: 2,
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
});

test('filter prop', () => {
  const wrapper = mount(TimePicker, {
    props: {
      filter,
      modelValue: '12:00',
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
});

test('formatter prop', async () => {
  const wrapper = mount(TimePicker, {
    props: {
      filter,
      formatter,
      modelValue: '12:00',
    },
  });

  expect(wrapper.html()).toMatchSnapshot();

  triggerDrag(wrapper.find('.van-picker-column'), 0, -100);
  wrapper.find('.van-picker-column ul').trigger('transitionend');
  await later();

  expect((wrapper.vm as Record<string, any>).getPicker().getValues()).toEqual([
    '20 hour',
    '00 minute',
  ]);
});

test('confirm event', () => {
  const wrapper = mount(TimePicker, {
    props: {
      modelValue: '12:00',
    },
  });

  triggerDrag(wrapper.find('.van-picker-column'), 0, -100);
  wrapper.find('.van-picker__confirm').trigger('click');
  expect(wrapper.emitted<[string]>('confirm')![0][0]).toEqual('23:00');
});

test('cancel event', () => {
  const wrapper = mount(TimePicker);

  wrapper.find('.van-picker__cancel').trigger('click');
  expect(wrapper.emitted('cancel')).toBeTruthy();
});

test('dynamic set value', async () => {
  const wrapper = mount(TimePicker);

  await wrapper.setProps({ modelValue: '00:00' });
  wrapper.find('.van-picker__confirm').trigger('click');
  await wrapper.setProps({ modelValue: '22:30' });
  wrapper.find('.van-picker__confirm').trigger('click');

  expect(wrapper.emitted<[string]>('confirm')![0][0]).toEqual('00:00');
  expect(wrapper.emitted<[string]>('confirm')![1][0]).toEqual('22:30');
});

test('change min-minute and emit correct value', async () => {
  const wrapper = mount(TimePicker, {
    props: {
      modelValue: '12:00',
      minMinute: 0,
    },
  });

  await later();
  await wrapper.setProps({ minMinute: 30 });
  wrapper.find('.van-picker__confirm').trigger('click');
  expect(wrapper.emitted<[string]>('confirm')![0][0]).toEqual('12:30');
});

test('set max-hour & max-minute smaller than current then emit correct value', async () => {
  const wrapper = mount(TimePicker, {
    props: {
      modelValue: '23:59',
    },
  });

  await later();
  await wrapper.setProps({
    maxHour: 2,
    maxMinute: 2,
  });

  wrapper.find('.van-picker__confirm').trigger('click');
  expect(wrapper.emitted<[string]>('confirm')![0][0]).toEqual('00:00');
});

test('set min-minute dynamically', async () => {
  const wrapper = mount({
    emits: ['change'],
    setup(_, { emit }) {
      const currentTime = ref('12:30');
      return () => (
        <TimePicker
          v-model={currentTime.value}
          minMinute={Number(currentTime.value.split(':')[0]) > 12 ? 0 : 30}
          minHour={12}
          maxHour={13}
          onChange={(value) => emit('change', value)}
        />
      );
    },
  });

  triggerDrag(wrapper.find('.van-picker-column'), 0, -100);
  wrapper.find('.van-picker__confirm').trigger('click');
  await later();
  expect(wrapper.emitted<[string]>('change')![0][0]).toEqual('12:30');
});

test('should emit value correctly when dynamic change min-hour', async () => {
  const wrapper = mount({
    emits: ['confirm'],
    setup(_, { emit }) {
      const state = reactive({
        time: '10:30',
        minHour: 1,
      });

      const onChange = () => {
        state.minHour = 11;
      };

      useExpose({
        onChange,
      });

      return () => (
        <TimePicker
          v-model={state.time}
          minHour={state.minHour}
          onConfirm={(value: Date) => emit('confirm', value)}
        />
      );
    },
  });

  await later();
  (wrapper.vm as Record<string, any>).onChange();
  await later();
  wrapper.find('.van-picker__confirm').trigger('click');
  expect(wrapper.emitted<[Date]>('confirm')![0][0]).toEqual('11:30');
});
