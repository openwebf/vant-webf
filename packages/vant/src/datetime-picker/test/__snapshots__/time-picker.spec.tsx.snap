// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filter prop 1`] = `
<div class="van-picker">
  <div class="van-picker__toolbar">
    <button type="button"
            class="van-picker__cancel van-haptics-feedback"
    >
      Cancel
    </button>
    <button type="button"
            class="van-picker__confirm van-haptics-feedback"
    >
      Confirm
    </button>
  </div>
  <div class="van-picker__columns"
       style="height: 264px;"
  >
    <div class="van-picker-column">
      <ul style="transform: translate3d(0, 110px, 0); transition-duration: 0ms; transition-property: none;"
          class="van-picker-column__wrapper"
      >
        <li role="button"
            style="height: 44px;"
            tabindex="0"
            class="van-picker-column__item van-picker-column__item--selected"
        >
          <div class="van-ellipsis">
            00
          </div>
        </li>
        <li role="button"
            style="height: 44px;"
            tabindex="0"
            class="van-picker-column__item"
        >
          <div class="van-ellipsis">
            05
          </div>
        </li>
        <li role="button"
            style="height: 44px;"
            tabindex="0"
            class="van-picker-column__item"
        >
          <div class="van-ellipsis">
            10
          </div>
        </li>
        <li role="button"
            style="height: 44px;"
            tabindex="0"
            class="van-picker-column__item"
        >
          <div class="van-ellipsis">
            15
          </div>
        </li>
        <li role="button"
            style="height: 44px;"
            tabindex="0"
            class="van-picker-column__item"
        >
          <div class="van-ellipsis">
            20
          </div>
        </li>
      </ul>
    </div>
    <div class="van-picker-column">
      <ul style="transform: translate3d(0, 110px, 0); transition-duration: 0ms; transition-property: none;"
          class="van-picker-column__wrapper"
      >
        <li role="button"
            style="height: 44px;"
            tabindex="0"
            class="van-picker-column__item van-picker-column__item--selected"
        >
          <div class="van-ellipsis">
            00
          </div>
        </li>
        <li role="button"
            style="height: 44px;"
            tabindex="0"
            class="van-picker-column__item"
        >
          <div class="van-ellipsis">
            10
          </div>
        </li>
        <li role="button"
            style="height: 44px;"
            tabindex="0"
            class="van-picker-column__item"
        >
          <div class="van-ellipsis">
            20
          </div>
        </li>
        <li role="button"
            style="height: 44px;"
            tabindex="0"
            class="van-picker-column__item"
        >
          <div class="van-ellipsis">
            30
          </div>
        </li>
        <li role="button"
            style="height: 44px;"
            tabindex="0"
            class="van-picker-column__item"
        >
          <div class="van-ellipsis">
            40
          </div>
        </li>
        <li role="button"
            style="height: 44px;"
            tabindex="0"
            class="van-picker-column__item"
        >
          <div class="van-ellipsis">
            50
          </div>
        </li>
      </ul>
    </div>
    <div class="van-picker__mask"
         style="background-size: 100% 110px;"
    >
    </div>
    <div class="van-hairline-unset--top-bottom van-picker__frame"
         style="height: 44px;"
    >
    </div>
  </div>
</div>
`;

exports[`format initial value 1`] = `
<div class="van-picker">
  <div class="van-picker__toolbar">
    <button type="button"
            class="van-picker__cancel van-haptics-feedback"
    >
      Cancel
    </button>
    <button type="button"
            class="van-picker__confirm van-haptics-feedback"
    >
      Confirm
    </button>
  </div>
  <div class="van-picker__columns"
       style="height: 264px;"
  >
    <div class="van-picker-column">
      <ul style="transform: translate3d(0, 110px, 0); transition-duration: 0ms; transition-property: none;"
          class="van-picker-column__wrapper"
      >
        <li role="button"
            style="height: 44px;"
            tabindex="0"
            class="van-picker-column__item van-picker-column__item--selected"
        >
          <div class="van-ellipsis">
            22
          </div>
        </li>
        <li role="button"
            style="height: 44px;"
            tabindex="0"
            class="van-picker-column__item"
        >
          <div class="van-ellipsis">
            23
          </div>
        </li>
      </ul>
    </div>
    <div class="van-picker-column">
      <ul style="transform: translate3d(0, 110px, 0); transition-duration: 0ms; transition-property: none;"
          class="van-picker-column__wrapper"
      >
        <li role="button"
            style="height: 44px;"
            tabindex="0"
            class="van-picker-column__item van-picker-column__item--selected"
        >
          <div class="van-ellipsis">
            58
          </div>
        </li>
        <li role="button"
            style="height: 44px;"
            tabindex="0"
            class="van-picker-column__item"
        >
          <div class="van-ellipsis">
            59
          </div>
        </li>
      </ul>
    </div>
    <div class="van-picker__mask"
         style="background-size: 100% 110px;"
    >
    </div>
    <div class="van-hairline-unset--top-bottom van-picker__frame"
         style="height: 44px;"
    >
    </div>
  </div>
</div>
`;

exports[`formatter prop 1`] = `
<div class="van-picker">
  <div class="van-picker__toolbar">
    <button type="button"
            class="van-picker__cancel van-haptics-feedback"
    >
      Cancel
    </button>
    <button type="button"
            class="van-picker__confirm van-haptics-feedback"
    >
      Confirm
    </button>
  </div>
  <div class="van-picker__columns"
       style="height: 264px;"
  >
    <div class="van-picker-column">
      <ul style="transform: translate3d(0, 110px, 0); transition-duration: 0ms; transition-property: none;"
          class="van-picker-column__wrapper"
      >
        <li role="button"
            style="height: 44px;"
            tabindex="0"
            class="van-picker-column__item van-picker-column__item--selected"
        >
          <div class="van-ellipsis">
            00 hour
          </div>
        </li>
        <li role="button"
            style="height: 44px;"
            tabindex="0"
            class="van-picker-column__item"
        >
          <div class="van-ellipsis">
            05 hour
          </div>
        </li>
        <li role="button"
            style="height: 44px;"
            tabindex="0"
            class="van-picker-column__item"
        >
          <div class="van-ellipsis">
            10 hour
          </div>
        </li>
        <li role="button"
            style="height: 44px;"
            tabindex="0"
            class="van-picker-column__item"
        >
          <div class="van-ellipsis">
            15 hour
          </div>
        </li>
        <li role="button"
            style="height: 44px;"
            tabindex="0"
            class="van-picker-column__item"
        >
          <div class="van-ellipsis">
            20 hour
          </div>
        </li>
      </ul>
    </div>
    <div class="van-picker-column">
      <ul style="transform: translate3d(0, 110px, 0); transition-duration: 0ms; transition-property: none;"
          class="van-picker-column__wrapper"
      >
        <li role="button"
            style="height: 44px;"
            tabindex="0"
            class="van-picker-column__item van-picker-column__item--selected"
        >
          <div class="van-ellipsis">
            00 minute
          </div>
        </li>
        <li role="button"
            style="height: 44px;"
            tabindex="0"
            class="van-picker-column__item"
        >
          <div class="van-ellipsis">
            10 minute
          </div>
        </li>
        <li role="button"
            style="height: 44px;"
            tabindex="0"
            class="van-picker-column__item"
        >
          <div class="van-ellipsis">
            20 minute
          </div>
        </li>
        <li role="button"
            style="height: 44px;"
            tabindex="0"
            class="van-picker-column__item"
        >
          <div class="van-ellipsis">
            30 minute
          </div>
        </li>
        <li role="button"
            style="height: 44px;"
            tabindex="0"
            class="van-picker-column__item"
        >
          <div class="van-ellipsis">
            40 minute
          </div>
        </li>
        <li role="button"
            style="height: 44px;"
            tabindex="0"
            class="van-picker-column__item"
        >
          <div class="van-ellipsis">
            50 minute
          </div>
        </li>
      </ul>
    </div>
    <div class="van-picker__mask"
         style="background-size: 100% 110px;"
    >
    </div>
    <div class="van-hairline-unset--top-bottom van-picker__frame"
         style="height: 44px;"
    >
    </div>
  </div>
</div>
`;

exports[`max-hour & max-minute 1`] = `
<div class="van-picker">
  <div class="van-picker__toolbar">
    <button type="button"
            class="van-picker__cancel van-haptics-feedback"
    >
      Cancel
    </button>
    <button type="button"
            class="van-picker__confirm van-haptics-feedback"
    >
      Confirm
    </button>
  </div>
  <div class="van-picker__columns"
       style="height: 264px;"
  >
    <div class="van-picker-column">
      <ul style="transform: translate3d(0, 110px, 0); transition-duration: 0ms; transition-property: none;"
          class="van-picker-column__wrapper"
      >
        <li role="button"
            style="height: 44px;"
            tabindex="0"
            class="van-picker-column__item van-picker-column__item--selected"
        >
          <div class="van-ellipsis">
            00
          </div>
        </li>
        <li role="button"
            style="height: 44px;"
            tabindex="0"
            class="van-picker-column__item"
        >
          <div class="van-ellipsis">
            01
          </div>
        </li>
        <li role="button"
            style="height: 44px;"
            tabindex="0"
            class="van-picker-column__item"
        >
          <div class="van-ellipsis">
            02
          </div>
        </li>
      </ul>
    </div>
    <div class="van-picker-column">
      <ul style="transform: translate3d(0, 110px, 0); transition-duration: 0ms; transition-property: none;"
          class="van-picker-column__wrapper"
      >
        <li role="button"
            style="height: 44px;"
            tabindex="0"
            class="van-picker-column__item van-picker-column__item--selected"
        >
          <div class="van-ellipsis">
            00
          </div>
        </li>
        <li role="button"
            style="height: 44px;"
            tabindex="0"
            class="van-picker-column__item"
        >
          <div class="van-ellipsis">
            01
          </div>
        </li>
        <li role="button"
            style="height: 44px;"
            tabindex="0"
            class="van-picker-column__item"
        >
          <div class="van-ellipsis">
            02
          </div>
        </li>
      </ul>
    </div>
    <div class="van-picker__mask"
         style="background-size: 100% 110px;"
    >
    </div>
    <div class="van-hairline-unset--top-bottom van-picker__frame"
         style="height: 44px;"
    >
    </div>
  </div>
</div>
`;
